breakpad-symbols/src/sym_file/mod.rs

```diff
@@ -128,6 +150,26 @@ impl SymbolFile {
                 if fully_consumed {
                     return Ok(parser.finish());
                 } else {
+                    // The reader is exhausted, but the parser was not fully
+                    // consumed on the last iteration. Probably because it was
+                    // expecting a trailing newline. Lets give it one, and do
+                    // one more parse attempt.
+                    let space = buf.space();
+                    if !space.is_empty() {
+                        space[0] = b'\n';
+                        buf.fill(1);
+
+                        let input = buf.data();
+                        let consumed = parser.parse_more(input)?;
+                        if consumed > 0 {
+                            callback(&input[..consumed]);
```

Comment:
hmm this part is mildly concerning. I think if the symbol file is missing the newline, we should preserve that fact in the cache, and just rely on the parser itself always handling it every time it's loaded?

Reply:
hm, I’m not sure I follow? I used a `- 1` here, to make sure it matches the original stream.